WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/SiteCreationDomainsViewModel.kt

```diff
@@ -233,21 +243,30 @@ class SiteCreationDomainsViewModel @Inject constructor(
 
     private fun createSuggestionsUiStates(
         onRetry: () -> Unit,
+        query: String?,
         data: List<String>,
         errorFetchingSuggestions: Boolean,
         @StringRes errorResId: Int?
     ): List<DomainsListItemUiState> {
         val items: ArrayList<DomainsListItemUiState> = ArrayList()
         if (errorFetchingSuggestions) {
             val errorUiState = DomainsFetchSuggestionsErrorUiState(
-                    messageResId = errorResId ?: R.string.site_creation_fetch_suggestions_error_unknown,
+                    messageResId = errorResId
+                            ?: R.string.site_creation_fetch_suggestions_error_unknown,
                     retryButtonResId = R.string.button_retry
             )
             errorUiState.onItemTapped = onRetry
             items.add(errorUiState)
         } else {
+            if (data.isNotEmpty())
```

Comment:
Are we missing `{` here? I assume we want to wrap both `query?.let` and `data.forEach`, right? In case you really want to wrap just `query?.let`, I'd suggest adding the `{` explicitly - we try to avoid using `ifs` without specifying a block explicitly, as it leads to hard to notice bugs. Wdyt?

Reply:
Thanks for spotting this. Didn't realize there wasn't a brace. I will put them. 

> Are we missing { here? I assume we want to wrap both query?.let and data.forEach, right? 

Not necessarily. I put this there so the query doesn't run a check against an empty set of domains because if it does then it won't find it and will show the unavailability state which would be incorrect since it wasn't compared against an actual result. Before doing this check the unavailability state would be visible while the suggestions are being loaded.

Reply:
Thanks for the clarification;). I'm wondering if it would be better to move it to the `getDomainUnavailableUiState` since it's basically a logic related to it. Wdyt?

Reply:
Thanks that's a good suggestion. Will do the move!

Reply:
Resolved here 9efa053